copy/copy.go

```diff
@@ -646,6 +688,27 @@ func (c *copier) copyOneImage(ctx context.Context, policyContext *signature.Poli
 	// If encrypted and decryption keys provided, we should try to decrypt
 	ic.diffIDsAreNeeded = ic.diffIDsAreNeeded || (isEncrypted(src) && ic.c.ociDecryptConfig != nil) || ic.c.ociEncryptConfig != nil
 
+	options.OptimizeDestinationImageAlreadyExists = true // TODO: PR: Remove forced enablement for PR testing
+	// If enabled, fetch and compare the destination's manifest. And as an optimization skip updating the destination iff equal
+	// To be enabled, options.OptimizeDestinationImageAlreadyExists must be true, the image must not be signed or encryted. And must
+	// not have any pending manifest updates
+	isSigned := len(sigs) > 0 || options.SignBy != "" // TODO: Consider allowing signatures updates only and skipping the image's layers/manifest copy if possible
+
+	logrus.Debugf("Evaluating image manifest check optimzation: options.OptimizeDestinationImageAlreadyExists=%t !isSigned=%t !destRequiresOciEncryption=%t ic.noPendingManifestUpdates()=%t", options.OptimizeDestinationImageAlreadyExists, !isSigned, !destRequiresOciEncryption, ic.noPendingManifestUpdates())
```

Comment:
That debug line could be polished a bit — e.g. probably log `isSigned` instead of `!isSigned`, and the like.